src/Controller/HybridAuthController.php

```diff
@@ -76,10 +75,14 @@ public static function create(ContainerInterface $container) {
   public function authenticate($provider_id) {
     // Make sure the session is started, HybridAuth library needs it.
     // @todo find the way how to avoid session_start().
+    // https://symfony.com/doc/current/components/http_foundation/sessions.html ?
     session_start();
 
     try {
-      $provider = new LinkedIn($this->getConfiguration());
+      // Configure function name and path for it.
+      $provider_function = 'Hybridauth\\Provider\\' . $provider_id;
+      // Get provider from variable function.
+      $provider = new $provider_function($this->getConfiguration($provider_id));
```

Comment:
some checks like class_exists have to be added

Reply:
Added condition class_exists().